OsmAnd/src/net/osmand/plus/settings/fragments/PositionAnimationFragment.java

```diff
@@ -0,0 +1,103 @@
+package net.osmand.plus.settings.fragments;
+
+import static net.osmand.plus.utils.UiUtilities.CompoundButtonType.TOOLBAR;
+
+import android.content.Context;
+import android.graphics.drawable.ColorDrawable;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.TextView;
+
+import androidx.annotation.LayoutRes;
+import androidx.annotation.NonNull;
+import androidx.appcompat.widget.SwitchCompat;
+import androidx.preference.Preference;
+import androidx.preference.PreferenceScreen;
+import androidx.preference.PreferenceViewHolder;
+
+import net.osmand.plus.R;
+import net.osmand.plus.helpers.AndroidUiHelper;
+import net.osmand.plus.settings.preferences.PositionAnimationPreference;
+import net.osmand.plus.utils.AndroidUtils;
+import net.osmand.plus.utils.ColorUtilities;
+import net.osmand.plus.utils.UiUtilities;
+
+public class PositionAnimationFragment extends BaseSettingsFragment {
+
+	@Override
+	protected void setupPreferences() {
+		Context context = getContext();
+		PreferenceScreen screen = getPreferenceScreen();
+		if (context != null && screen != null) {
+			if (isPositionAnimationEnabled()) {
+				screen.addPreference(createPositionAnimationCard(context));
+			} else {
+				screen.addPreference(createPreference(context,
+						R.layout.card_position_animation_empty_banner));
+			}
```

Comment:
these preferences could be added in R.xml.position_animation_settings & in fragment only update their visibility/content